core/modules/filter/filter.pages.inc

```diff
@@ -77,6 +78,72 @@ function filter_format_editor_link_form($form, &$form_state, $format) {
 }
 
 /**
+ * Autocomplete callback for link textfield.
+ */
+function _filter_link_autocomplete($string) {
+  $matches = array();
+  if ($string) {
+    // Get node matches.
+    $results = db_select('node')
+      ->fields('node', array('nid', 'title'))
+      ->condition('title', db_like($string) . '%', 'LIKE')
+      ->addTag('node_access')
+      ->range(0, 10)
+      ->execute();
+    foreach ($results as $result) {
+      $node = node_load($result->nid);
```

Comment:
If the `node_load()` (or `user_load()` or `taxonomy_term_load()`) in unavoidable, we should at least use `node_load_multiple()` or equivalent.

You could also try to use an EntityFieldQuery to query against any type of entity, then use `entity_load_multiple()` to handle this in a generic fashion, but I don't think there's any generic way of selecting based on an entity's label (e.g. $node->title, $user->name, or $term->name), so I don't think that would be practical.